src/cljs/org/broadinstitute/firecloud_ui/page/workspace/analysis/track_selector.cljs

```diff
@@ -0,0 +1,156 @@
+                                                   (range (count tracks)))
+                                     closest-div-index (:index (apply min-key #(js/Math.abs (- y (:midpoint %))) div-locs))]
+                                 (when-not (= (:drop-index @state) closest-div-index)
+                                   (swap! state assoc :drop-index closest-div-index)))))}
+        [:div {:style {:margin "1em 0 0 1em"}}
+         "Selected tracks"]
+        (when (empty? tracks)
+          [:div {:style {:margin "1em" :textAlign "center"
+                         :border style/standard-line :borderRadius 4 :padding "1em"}}
+           "No tracks selected"])
+        [:div {:style {:marginTop "1em" :overflowY "auto"}}
+         (map-indexed
+           (fn [index {:keys [track-url index-url requires-index?] :as track}]
+             (if (= track :dummy)
+               [:div {:ref (str "track" index)
+                      :style {:display "flex" :alignItems "center" :justifyContent "center"
+                              :height 46 :padding "0 4px" :margin "2px 3px"
+                              :border (str "1px dotted " (:line-gray style/colors)) :borderRadius 4}}
+                (style/left-ellipses {} (:drag-url @state))]
+               [:div {:ref (str "track" index)
+                      :style {:display "flex" :alignItems "center" :padding 4 :margin "2px 3px"
+                              :border style/standard-line :borderRadius 4}}
+                (when (> (count tracks) 1)
+                  [:img {:src "assets/drag_temp.png"
+                         :style {:flex "0 0 auto" :height 16 :cursor "ns-resize"}
+                         :draggable false
+                         :onMouseDown #(swap! state assoc :drag-index index :drop-index index :drag-url track-url
+                                              :text-selection (common/disable-text-selection))}])
+                (case @index-url
+                  :pending [:div {} [comps/Spinner {:height "1em" :text "Searching for index file..."}]]
+                  :error (if requires-index?
+                           [:div {:style {:flex "1 1 auto" :marginLeft 8 :color (:exception-red style/colors) :overflow "hidden"}}
+                            (style/right-ellipses {} (str "Unable to find index file for '" track-url "'"))
+                            (style/right-ellipses {} "Please ensure you have the associated .bai or .bam.bai at the same path")]
+                           [:div {:style {:flex "1 1 auto" :marginLeft 8 :overflow "hidden"}}
+                            (style/left-ellipses {} track-url)
+                            (style/right-ellipses {} "Optional index file not found.")])
+                  [:div {:style {:flex "1 1 auto" :marginLeft 8 :overflow "hidden"}}
+                   (style/left-ellipses {} track-url)
+                   (style/left-ellipses {} @index-url)])
```

Comment:
tweak: might be nice to visually differentiate the index and track urls inside the track selector.